lib/services/app.go

```diff
@@ -158,3 +169,217 @@ func UnmarshalAppServer(data []byte, opts ...MarshalOption) (types.AppServer, er
 	}
 	return nil, trace.BadParameter("unsupported app server resource version %q", h.Version)
 }
+
+// NewApplicationsFromKubeService creates application resources from kubernetes service. One service result
+// in multiple application resources if there are multiple ports exposed.
+func NewApplicationsFromKubeService(service v1.Service, clusterName string, pi ProtocolChecker) ([]types.Application, error) {
```

Comment:
nit: it's called `pc` elsewhere
```suggestion
func NewApplicationsFromKubeService(service v1.Service, clusterName string, pc ProtocolChecker) ([]types.Application, error) {
```

Reply:
Yep, remnants of time when interface was called `protocolInspector` 😅